pipeline.Makefile

```diff
@@ -109,7 +117,34 @@ schema-help:
 	@:$(info =================)
 	@:$(info $(DEBUG))
 
+.PHONY: validate-schema
+validate-schema: $(SCHEMA_FILE)
+	@mkdir -p $(VALIDATOR_OUTPUT_DIR)
+	@echo "Validating schema $(SCHEMA_FILE)..."
+	@$(RUN) linkml validate --schema $< > $(SCHEMA_VALIDATE_LOG) 2>&1 && \
+		echo "Schema validation passed." >> $(SCHEMA_VALIDATE_LOG) || \
+		echo "Schema validation failed." >> $(SCHEMA_VALIDATE_LOG)
+	@echo "  Log written to $(SCHEMA_VALIDATE_LOG)"
+
 
-# .PHONY: validate-schema
-# validate: $(SCHEMA_FILE)
-# 	$(RUN) linkml-validate -s $< -C $(INPUT_FILES)
+.PHONY: validate-data
+validate-data: $(SCHEMA_FILE)
+	@:$(call check_input_files)
+	@mkdir -p $(VALIDATOR_OUTPUT_DIR)
+	@rm -f $(DATA_VALIDATE_LOG)
+	@for f in $(INPUT_FILES); do \
+		class=$$(basename $$f | sed -E 's/\.[ct]sv$$//' | tr '-' '_' | tr '[:upper:]' '[:lower:]'); \
```

Comment:
What would you think about having a variable in make called `GENERATED_CLASSES` that we define right after `INPUT_FILES`? It could be useful for debugging. (Or if we were to use classes in another recipe). That might look like:

```
GENERATED_CLASSES := $(shell echo $(basename $(notdir $(INPUT_FILES))) | tr '-' '_' | tr '[:upper:]' '[:lower:]')
```

(Note that this `basename` in that example is a make function that [strips suffixes](https://www.gnu.org/software/make/manual/html_node/File-Name-Functions.html#index-basename-1), rather than the coreutils program `basename` which is used in the code I'm commenting on, which doesn't).